src/Microsoft.AspNetCore.Mvc.Core/ConsumesAttribute.cs

```diff
@@ -184,6 +192,83 @@ public bool Accept(ActionConstraintContext context)
             return true;
         }
 
+        /// <inheritdoc />
+        public bool Accept(EndpointConstraintContext context)
+        {
+            // If this constraint is not closest to the action, it will be skipped.
+            if (!IsApplicable(context.CurrentCandidate.Endpoint))
+            {
+                // Since the constraint is to be skipped, returning true here
+                // will let the current candidate ignore this constraint and will
+                // be selected based on other constraints for this action.
+                return true;
+            }
+
+            var requestContentType = context.HttpContext.Request.ContentType;
+
+            // If the request content type is null we need to act like pass through.
+            // In case there is a single candidate with a constraint it should be selected.
+            // If there are multiple actions with consumes action constraints this should result in ambiguous exception
+            // unless there is another action without a consumes constraint.
+            if (requestContentType == null)
+            {
+                var isActionWithoutConsumeConstraintPresent = context.Candidates.Any(
+                    candidate => candidate.Constraints == null ||
+                    !candidate.Constraints.Any(constraint => constraint is IConsumesEndpointConstraint));
+
+                return !isActionWithoutConsumeConstraintPresent;
```

Comment:
Is most of this just copy-paste?

This kind of logic is something we'll have to think about - though I'm not sure if we did it anywhere else than consumes.

Reply:
For instance we might handle this in MVC by creating a 'sentinel' endpoint for this case that always returns a 415.

Or we might build a first class feature into dispatcher for this 🤔

Reply:
It's copy-paste.

Reply:
I'm going to change `action` -> `endpoint`. Are there other changes you want to make here in the future? Do we want an issue?